skycfg_test.go

```diff
@@ -349,7 +389,12 @@ func TestSkycfgEndToEnd(t *testing.T) {
 	}
 
 	fnExecSkycfg := ExecSkycfg(func(config *skycfg.Config, testCase endToEndTestCase) ([]proto.Message, error) {
-		return config.Main(context.Background(), skycfg.WithVars(testCase.vars))
+		if testCase.flattenLists {
+			return config.Main(context.Background(), skycfg.WithVars(testCase.vars), skycfg.WithFlattenLists())
+		} else {
+			return config.Main(context.Background(), skycfg.WithVars(testCase.vars))
+		}
+
```

Comment:
Can we make the `flattenLists` property in the test cases an execOptions slice. We can then provide `skycfg.WithFlattenLists()` as a value in the test case and pass in the `testCase.execOptions...` in the call and forego the if/else statement.

```go
type endToEndTestCase struct {
        ...
	execOptions []skycfg.ExecOption
}


testCases := []endToEndTestCase{
	...
	endToEndTestCase{
			caseName:   "flatten nested list",
			...
			execOptions: []skycfg.ExecOption{skycfg.WithFlattenLists()},
	},
}


return config.Main(context.Background(), skycfg.WithVars(testCase.vars), testCase.execOptions...)
```